packages/store/src/defineHeapKind.js

```diff
@@ -0,0 +1,72 @@
+// @ts-check
+
+import { Far } from '@endo/marshal';
+
+const { ownKeys } = Reflect;
+
+const becomePassable = (tag, thing) => {
+  const names = ownKeys(thing);
+  if (names.length === 0 || typeof thing[names[0]] === 'function') {
+    return Far(tag, thing);
+  }
+  for (const name of names) {
+    assert.typeof(name, 'string');
+    Far(`${tag} ${name}`, thing[name]);
+  }
+  return harden(thing);
+};
+harden(becomePassable);
+
+/**
+ * Ersatz defineKind until virtual objects system is stable.
+ *
+ * @param {string} tag
+ * @param {Function} init
+ * @param {Function} actualize
+ * @param {Function} [finish]
+ */
+export const defineHeapKind = (tag, init, actualize, finish) => {
+  let nextInstanceID = 1;
+  const propertyNames = new Set();
+
+  function makeRepresentative(innerSelf) {
+    const wrappedData = {};
+    for (const prop of propertyNames) {
+      Object.defineProperty(wrappedData, prop, {
+        get: () => {
+          return innerSelf.rawData[prop];
+        },
+        set: value => {
+          innerSelf.rawData[prop] = value;
+        },
+      });
+    }
+    harden(wrappedData);
```

Comment:
Note that even though you're wrapping the data with getters and setters, you're not passing them through the marshaller and so you're not getting any of the arms-length handling that gives you.  In particular, you can set the value of a property and then mutate it in place, which you can't do with the other kinds of virtual objects.  If the goal is to emulate the constraints of the real thing so that code can be implemented to the real API, then you should probably at least be doing something to emulate the immutability of the values stored in virtual object properties.  I'm not sure how you'd do that, but if you don't then somebody is going to be surprised when they write something like `state.foo.push(47)` and have it work using `defineHeapKind` only to have it blow up in their faces when they switch to `defineKind`.

Reply:
It strikes me that what is implemented by SwingSet's test tooling (`fakeVirtualSupport.js` et al) is an always up to date full faith reproduction except in RAM, because it uses the actual virtual object manager only backed by a JavaScript `Map` rather than by the database.  You can't quite get away with using it directly since it also makes its own copy of LiveSlots and you'd want to use the real LiveSlots, but it's possible it could be modified to separate the storage so it can be backed by a `Map` even when the rest of your vat is saving stuff to disk.  Or at least it might be a starting point for a less brittle emulation.

Reply:
On the other hand, if you don't mind having mutable state properties, then you can probably simplify things by ditching the getter/setter wrapper.